src/fragmentarium/ui/search/ReferenceSearchForm.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react'
+import { Col, Form, Row } from 'react-bootstrap'
+import BibliographySelect from './BibliographySelect'
+
+type Props = {
+  fragmentService
+  onChange(searchForm: string, searchQuery: string): void
+  getState(key: string): string
+}
+
+class ReferenceSearchForm extends Component<Props> {
+  onChangePage = (event) => {
+    this.props.onChange('pages', event.target.value || '')
```

Comment:
You should only pass the value and not couple the implementation with something on the outside. Whoever defines the callback should make do the correct thing without the form knowing its internals.

Reply:
I fixed this by defining multiple callbacks one for NumberSearchForm, ReferenceSearchForm and TransliterationForm.